datamap/mv/core/src/main/scala/org/apache/carbondata/mv/extension/MVParser.scala

```diff
@@ -0,0 +1,226 @@
+   */
+  private def carbonKeyWord(keys: String): Regex = {
+    ("(?i)" + keys).r
+  }
+
+  implicit def regexToParser(regex: Regex): Parser[String] = {
+    import lexical.Identifier
+    acceptMatch(
+      s"identifier matching regex ${ regex }",
+      { case Identifier(str) if regex.unapplySeq(str).isDefined => str }
+    )
+  }
+
+  // By default, use Reflection to find the reserved words defined in the sub class.
+  // NOTICE, Since the Keyword properties defined by sub class, we couldn't call this
+  // method during the parent class instantiation, because the sub class instance
+  // isn't created yet.
+  protected lazy val reservedWords: Seq[String] =
+  this
+    .getClass
+    .getMethods
+    .filter(_.getReturnType == classOf[Keyword])
+    .map(_.invoke(this).asInstanceOf[Keyword].normalize)
+
+  // Set the keywords as empty by default, will change that later.
+  override val lexical = new SqlLexical
+
+  protected case class Keyword(str: String) {
+    def normalize: String = lexical.normalizeKeyword(str)
+    def parser: Parser[String] = normalize
+  }
+
+  def parse(input: String): LogicalPlan = {
+    synchronized {
+      phrase(start)(new lexical.Scanner(input)) match {
+        case Success(plan, _) =>
+          plan
+        case failureOrError =>
+          CarbonException.analysisException(failureOrError.toString)
+      }
+    }
+  }
+
+  private lazy val start: Parser[LogicalPlan] = mvCommand
+
+  private lazy val mvCommand: Parser[LogicalPlan] =
+    createMV | dropMV | showMV | rebuildMV | compactMV
+
+  /**
+   * CREATE MATERIALIZED VIEW IF NOT EXISTS mv_name
+   * MVPROPERTIES('KEY'='VALUE') AS mv_query_statement
+   */
+  private lazy val createMV: Parser[LogicalPlan] =
+    CREATE ~> MATERIALIZED ~> VIEW ~> opt(IF ~> NOT ~> EXISTS) ~ ident ~
+    opt(WITH ~> DEFERRED ~> REBUILD) ~
+    (MVPROPERTIES ~> "(" ~> repsep(options, ",") <~ ")").? ~
+    (AS ~> restInput).? <~ opt(";") ^^ {
+      case ifNotExists ~ mvName ~ deferredRebuild ~ mvProperties ~ query =>
+        val map = mvProperties.getOrElse(List[(String, String)]()).toMap[String, String]
+        CarbonCreateDataMapCommand(mvName, None, "mv", map, query,
+          ifNotExists.isDefined, deferredRebuild.isDefined)
+    }
+
+  /**
+   * DROP MATERIALIZED VIEW IF EXISTS mv_name
+   */
+  private lazy val dropMV: Parser[LogicalPlan] =
+    DROP ~> MATERIALIZED ~> VIEW ~> opt(IF ~> EXISTS) ~ ident <~ opt(";") ^^ {
+      case ifExits ~ mvName =>
+        CarbonDropDataMapCommand(mvName, ifExits.isDefined, None)
+    }
+
+  /**
+   * SHOW MATERIALIZED VIEWS
+   */
+  private lazy val showMV: Parser[LogicalPlan] =
+    SHOW ~> MATERIALIZED ~> VIEWS ~> opt(onTable) <~ opt(";") ^^ {
+      case tableIdent =>
+        CarbonDataMapShowCommand(tableIdent)
+    }
+
+  /**
+   * REBUILD MATERIALIZED VIEW mv_name
+   */
+  private lazy val rebuildMV: Parser[LogicalPlan] =
+    REBUILD ~> MATERIALIZED ~> VIEW ~> ident <~ opt(";") ^^ {
```

Comment:
```suggestion
    REBUILD ~> MATERIALIZED ~> VIEW ~> ident ~ ONTABLE ~ (ident <~ ".").? ~ ident<~ opt(";") ^^ {
```

Reply:
On table is not required because MV can be a join, so I think better not forcing user to give this